src/DocumentFormat.OpenXml/Packaging/OpenXmlPackage.FlatOpc.cs

```diff
@@ -130,11 +130,26 @@ private static XElement GetBinaryPartContentsAsXml(PackagePart part)
         private static string ToChunkedBase64String(PackagePart part)
         {
             using var stream = part.GetStream();
-            using var memory = new MemoryStream((int)stream.Length);
+            var size = checked((int)stream.Length);
+            var buffer = new byte[size];
 
-            stream.CopyTo(memory);
+#if NET7_0_OR_GREATER
+            stream.ReadExactly(buffer);
+#else
+            var totalRead = 0;
```

Comment:
Could you move this to an internal extension method so that call site here just calls `ReadExactly`? The `common/` folder has a bunch of these polyfills where that could live

Reply:
@twsouthwick
I can add a `StreamReadExtensions` file with the `ReadExactly` shim, but this file should be included only on NET6.0 or below, and I have no idea how it could be nicely done in [this shim targets file](https://github.com/OfficeDev/Open-XML-SDK/blob/main/src/common/FrameworkShims.targets) 😅

Reply:
I'm fine with doing either:

1) Create the file with an #if/#def around it from `#if !NET7_0_OR_GREATER` and include it by default in a new itemgroup with no conditions
2) Create the file and add it into all the sections for applicable TFMs, *and* create a new `<ItemGroup Condition=" '$(TargetFramework)' == 'net6.0'"/>`

I've been rethinking how to do the polyfills and am not convinced the current way is ideal (it proved harder than I wanted to update things to support builds for .net 6 and .net 7).